packages/shared/src/util/features.ts

```diff
@@ -114,3 +116,54 @@ export function validateFeatureValue(
 
   return value;
 }
+
+type StaleFeatureReason =
+  | "error"
+  | "draft-state"
+  | "no-active-envs"
+  | "no-rules"
+  | "no-active-exps"
+  | "all-exps-onesided";
+
+export function isFeatureStale(
+  feature: FeatureInterface,
+  linkedExperiments: ExperimentInterface[] | undefined = []
+): { stale: boolean; reason?: StaleFeatureReason } {
+  if (feature.linkedExperiments?.length && !linkedExperiments.length) {
+    // eslint-disable-next-line no-console
+    console.error("isFeatureStale: linkedExperiments not provided");
+    return { stale: false, reason: "error" };
+  }
+
+  const twoWeeksAgo = subWeeks(new Date(), 2);
+  // TODO why is dateUpdated a string?
+  const stale = new Date(feature.dateUpdated) < twoWeeksAgo;
+
+  if (!stale) return { stale };
+
+  if (feature.draft) return { stale, reason: "draft-state" };
+
+  const envSettings = Object.values(feature.environmentSettings ?? {});
+
+  const noneEnabled = !envSettings.some((e) => e.enabled);
+  if (noneEnabled) return { stale, reason: "no-active-envs" };
+
+  const rules = envSettings.map((e) => e.rules).flat();
+
+  if (rules.length === 0) return { stale, reason: "no-rules" };
+
+  // TODO check if there are 'active' rules and return early if os
+
+  const noExpsActive = !linkedExperiments.some((e) => e.status === "running");
+  if (noExpsActive) return { stale, reason: "no-active-exps" };
+
+  const allExpsOneSided = linkedExperiments.every((e) => {
+    const latestPhase = e.phases.slice(-1)?.[0];
+    if (!latestPhase) return false;
+    return latestPhase.variationWeights.includes(1);
+  });
```

Comment:
I don't think we need this check.  This hardly ever happens and if the experiment is active (check above), we don't want to mark this as stale, even if the traffic split is one-sided.